src/app.rs

```diff
@@ -428,14 +430,26 @@ impl App {
                 [OutputComponent::Numbers].iter().cloned().collect()
             } else if matches.is_present("plain") {
                 [OutputComponent::Plain].iter().cloned().collect()
-            } else {
+            } else if matches.is_present("style") {
                 values_t!(matches.values_of("style"), OutputComponent)?
                     .into_iter()
                     .map(|style| style.components(self.interactive_output))
                     .fold(HashSet::new(), |mut acc, components| {
                         acc.extend(components.iter().cloned());
                         acc
                     })
+            } else {
+                let style = env::var("BAT_STYLE").unwrap_or(String::from(BAT_STYLE_DEFAULT));
+                match OutputComponent::from_str(&style) {
```

Comment:
It would be great if we could remove the code duplication here, yes.

Also, it looks like this will not work for something like `BAT_STYLE="numbers,header"`, right?

Reply:
I would put this code into the previously existing `else` branch and first build an iterator over the `OutputComponent`s (either from `env` or from the `--style` argument`). Afterwards, we can call `.map(..).fold(...)` etc.